src/software/geom/polynomial.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include <cmath>
+#include <vector>
+
+#include "software/geom/point.h"
+
+/**
+ * Polynomial is a representation of a polynomial
+ * of the form coeffs[0] + coeffs[1] * + coeff[2] * x^2...
+ */
+class Polynomial
+{
+   public:
+    Polynomial() = delete;
+    /**
+     * Construct a polynomial from coefficients
+     *
+     * @param coeffs coefficients of the polynomial
+     *
+     * @note will prune off coeffs to enforce invariant:
+     *  *coeffs[coeffs.size()-1] != 0
+     *  *order of polynomial is coeffs.size()
+     */
+    explicit Polynomial(const std::vector<double>& coeffs);
+
+    /**
+     * Construct a polynomial from coefficients
+     *
+     * @param coeffs coefficients of the polynomial
+     *
+     * @note will prune off coeffs to enforce invariant:
+     *  *coeffs[coeffs.size()-1] != 0
+     *  *order of polynomial is coeffs.size()
```

Comment:
What exactly does this mean? Also if there are more detailed notes for a jdoc function, they should go after the brief one-liner at the top of the jdoc and before with first `@param`. Same goes for the other places here.

Reply:
we don't want to have 0x^5 + 0x^4 + 0x^3 + 2x^2 +3 look like a 5th order polynomial

Reply:
What doe `order of polynomial is coeffs.size()` mean? I read that as the order is some integer value?